### PR TITLE
Max line length

### PR DESCRIPTION
**Pull request by mdczaplicki**
Small update to have max-line-length as a argument.

Sorry, but didn't have time to make tests pass. Some magic happens below :stuck_out_tongue: 

Closes https://github.com/DanielNoord/pydocstringformatter/issues/90

### CODE REVIEW

File: .gitignore
Comment (DanielNoord) [inline .gitignore:6]: I'd prefer if you could add these to your own (global) `gitignore`. We don't really add any directories that are specific to any IDE or local environment as those should ideally be ignored by the user.

File: docs/usage.rst
Comment (DanielNoord) [inline docs/usage.rst:20]: It seems the CI fails here. You could try:
```python
cd docs
make html
```

That should update the docs.

File: pydocstringformatter/configuration/arguments_manager.py
Comment (DanielNoord) [inline pydocstringformatter/configuration/arguments_manager.py:113]:
```suggestion
            action="store",
```

See `argparse` [documentation](https://docs.python.org/3/library/argparse.html#action). `store_const` actually stores one value `88` in this case and doesn't allow specifying another value. `store` should work for us!

File: pydocstringformatter/formatting/base.py
Comment (DanielNoord) [inline pydocstringformatter/formatting/base.py:148]: Pff, you found quite a bit of spelling errors 😄

File: requirements-test.txt
Comment (DanielNoord) [inline requirements-test.txt:8]: 👍

File: tests/data/format/max_line_length/function_docstrings.args
Comment (DanielNoord) [inline tests/data/format/max_line_length/function_docstrings.args:1]: I'm not sure this works. We might need:
```suggestion
--max-line-length=50
```

File: tests/data/format/max_line_length/function_docstrings.py
Comment (DanielNoord) [inline tests/data/format/max_line_length/function_docstrings.py:1]: Basically the test should have 3 cases:
1) One multi-line summary clearly above 50 characters. > Summary should be wrapped.
2) One multi-line summary with a description above 50 chars. > Summary should be wrapped, but description shouldn't as we don't support that.
3) One single-line docstring that is 49 chars without the final quotes. See `LinesChunk`.
If it is a single-line docstring we need to add the final quotes on the same line, so the line is not actually 49 but 52, so we should indeed wrap. That is what `LinesChunck` is testing.

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Thank you very much for taking this on @mdczaplicki. Very much appreciated!


I left some comments, let me know if you have any questions!

File: tests/data/format/max_line_length/function_docstrings.args
Comment (mdczaplicki) [reply tests/data/format/max_line_length/function_docstrings.args:1]: Oh, that's why :D

File: .gitignore
Comment (Pierre-Sassoulas) [reply .gitignore:6]: (You can define a global gitignore so you don't have to add this to all your gitignores) https://stackoverflow.com/a/7335487/2519059

File: pydocstringformatter/configuration/arguments_manager.py
Comment (mdczaplicki) [reply pydocstringformatter/configuration/arguments_manager.py:113]: Where do I set it to default 88 then?

Comment (DanielNoord) [reply pydocstringformatter/configuration/arguments_manager.py:113]:
```suggestion
            action="store",
            default=88,
            type=int,
```

See the example in the argparse docs:
```python
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--foo', default=42)
>>> parser.parse_args(['--foo', '2'])
Namespace(foo='2')
>>> parser.parse_args([])
Namespace(foo=42)
```

File: tests/data/format/max_line_length/function_docstrings.py
Comment (mdczaplicki) [reply tests/data/format/max_line_length/function_docstrings.py:1]: OK. Updated them. But for some reason it seems that `50` is not picked up in this test.

Comment (DanielNoord) [reply tests/data/format/max_line_length/function_docstrings.py:1]: I think this is because the option doesn't use `type`. You could add a breakpoint somewhere after `line_length = self.config.max_line_length` but I think `line_length` will now be `"50"` instead of `50`. This can be fixed by using `type=int` in the `add_argument` call.

File: docs/usage.rst
Comment (mdczaplicki) [reply docs/usage.rst]: Did that. Unfortunately pipelines still complain

File: tests/data/format/max_line_length/function_docstrings.py
Comment (mdczaplicki) [reply tests/data/format/max_line_length/function_docstrings.py:1]: It actually says that all docstrings are properly formatted. So it doesn't run

File: tests/data/format/linewrap_summary/max_line_length_50.args
Comment (DanielNoord) [inline tests/data/format/linewrap_summary/max_line_length_50.args:1]: @mdczaplicki I have pushed some changes. Turns out I was being very dumb.

Since line wrapping is still experimental it is not enabled by default. So for the test to run, it also needs this flag set.

File: pydocstringformatter/configuration/arguments_manager.py
Comment (DanielNoord) [inline pydocstringformatter/configuration/arguments_manager.py:117]:
```suggestion
            help="Maximum line length of docstrings.",
```

File: docs/usage.rst
Comment (DanielNoord) [inline docs/usage.rst:46]:
```suggestion
                            Maximum line length of docstrings.
```

**Review by DanielNoord — APPROVED** [review]
One final spelling change.

I'm also using this PR to test some changes to the primer. Sorry for the merge commit spam 😅

File: tests/data/format/linewrap_summary/max_line_length_50.args
Comment (mdczaplicki) [reply tests/data/format/linewrap_summary/max_line_length_50.args:1]: No problem. Thanks for the help here :D

Comment (DanielNoord) [reply tests/data/format/linewrap_summary/max_line_length_50.args:1]: No worries and thanks very much for doing the initial work. It's much easier as maintainer to build on a PR that's 90% done than starting from scratch.

Sorry for the spam you're getting right now. There is something wrong with one of our test runs but I can only test it on a PR opened by somebody that is not a contributor... I'll clean the commits in this PR up after I have fixed the issue and then merge this!

Comment (mdczaplicki) [reply tests/data/format/linewrap_summary/max_line_length_50.args:1]: Tests pass locally. :tada: 

BTW. I believe there is a problem with `make html`. I was running it, but pipelines complained.

Also:
```
❯ cd docs
❯ pip install -e .
Obtaining file:///home/marek/PycharmProjects/pydocstringformatter/docs
ERROR: file:///home/marek/PycharmProjects/pydocstringformatter/docs does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
❯ pip install -r requirements-doc.txt
Obtaining file:///home/marek/PycharmProjects/pydocstringformatter/docs (from -r requirements-doc.txt (line 1))
ERROR: file:///home/marek/PycharmProjects/pydocstringformatter/docs (from -r requirements-doc.txt (line 1)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

However it works when I go 
```
cd ..
``` 
and then
```
pip install -r docs/requirements-doc.txt
```

Comment (DanielNoord) [reply tests/data/format/linewrap_summary/max_line_length_50.args:1]: Yeah it should be:
```python
pip install -e .
cd docs
make html
```

This is due to how `readthedocs` install dependencies from root. So we need to install them from root as well...

**Comment by DanielNoord** [comment]
Thanks once again @mdczaplicki, also for prompting me to final fix our CI 🎉

**Pull request merged**
